### DISCUSSION TITLE
How can I sync files created in a node.js process to the main repo

### DISCUSSION THREAD

**Discussion by Kief5555** [vercel/community#1451] [Help] [answered]
### Summary

So I have a system in my project where a file will be created when the api is interacted with.

Problem is, I don't want the file gone because it is a special system

My question is, when this file is created, will it automaticly commit to the github repo? (Cause if not, I loose the data that is created)

### Additional information

This is my first time using vercel so please don't get mad

**Comment by amyegan** [comment, marked as answer]
Good question, Kief! Welcome to Vercel. We're happy to have you here 😄 

Files created in Serverless Functions are temporary and are not automatically committed to your GitHub repo. But you can set up your app to write files to a permanent storage service, such as your [git repo](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents) or an [AWS S3 bucket](https://vercel.com/guides/how-can-i-use-aws-s3-with-vercel). 

We have a guide about [using files in Serverless Functions on Vercel](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions) which includes code for an example [temporary storage](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage) scenario

**Reply by Kief5555** [reply]
Hey @amyegan! Thanks for the reply 🙂.


I looked through the link and it does not tell me how I can do it with Git 😅. Are you able to provide an example on how I can do this?

Thanks

**Reply by amyegan** [reply]
You could use the GitHub REST API to [create or update file contents](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents) by making a `POST` from your own serverless function API. I don't have an example of this, but you should be able to find the info you need in the [GitHub API docs](https://docs.github.com/en/rest?apiVersion=2022-11-28)
